File: src/app/api/responses/job-response.ts
import {FlagResponse} from './flag-response';
import {DescribedFlagResponse} from './described-flag-response';
import {SpecialityResponse} from './speciality-response';

export interface JobResponse {
    id: number;
    name: string;
    internalname?: string;
    informations?: string;
    playerDescription?: string;
    playerSummary?: string;
    maxLoad?: number;
    maxArmorPR?: number;
    isMagic?: boolean;
    baseEv?: number;
    factorEv?: number;
    bonusEv?: number;
    baseEa?: number;
    diceEaLevelUp?: number;
    baseAT?: number;
    basePRD?: number;
    parentJobId?: number;
    flags?: FlagResponse[];
    skillIds: number[];
    availableSkillIds: number[];
    originsWhitelist: { id: number, name: string }[];
    originsBlacklist: { id: number, name: string }[];
    bonuses: DescribedFlagResponse[];
    requirements: StatRequirementResponse[];
    restricts: DescribedFlagResponse[];
    specialities: SpecialityResponse[];
}

File: backend/src/Naheulbook.Web/Responses/JobResponse.cs
using JetBrains.Annotations;
using Naheulbook.Shared.TransientModels;

namespace Naheulbook.Web.Responses;


[PublicAPI]
public class JobResponse
{
    public Guid Id { get; set; }
    public string Name { get; set; } = null!;
    public string? Information { get; set; }

    public string? PlayerDescription { get; set; }
    public string? PlayerSummary { get; set; }

    public bool? IsMagic { get; set; }

    public JobData Data { get; set; } = null!;
    public List<FlagResponse>? Flags { get; set; }
    public List<Guid> SkillIds { get; set; } = null!;
    public List<Guid> AvailableSkillIds { get; set; } = null!;
    public List<DescribedFlagResponse> Bonuses { get; set; } = null!;
    public List<StatRequirementResponse> Requirements { get; set; } = null!;
    public List<DescribedFlagResponse> Restrictions { get; set; } = null!;
    public List<SpecialityResponse> Specialities { get; set; } = null!;
}